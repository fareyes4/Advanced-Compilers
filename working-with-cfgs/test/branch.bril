@main {
  c: bool = const true;
  br c l1 l2;
l1:
  jmp end;
l2:
  jmp end;
end:
  ret;
}
