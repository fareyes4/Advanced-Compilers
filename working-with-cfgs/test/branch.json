{
  "functions": [
    {
      "name": "main",
      "instrs": [
        { "op": "const", "type": "bool", "dest": "c", "value": true },
        { "op": "br", "args": ["c"], "labels": ["l1", "l2"] },
        { "label": "l1" },
        { "op": "jmp", "labels": ["end"] },
        { "label": "l2" },
        { "op": "jmp", "labels": ["end"] },
        { "label": "end" },
        { "op": "ret" }
      ]
    }
  ]
}
