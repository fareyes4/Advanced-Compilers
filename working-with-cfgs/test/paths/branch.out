_B0 0
l1 1
l2 1
end 2
